src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -0,0 +1,503 @@
+		const [variant, setVariant] = useState<BuildVariant>('prod');
+		const [showAllVersions, setShowAllVersions] = useState(false);
+		const { selectionOpts, preferredSelectionOpts } = useMemo(
+			() => getCategorizedVersions(osVersions, model.deviceType, osType),
+			[osVersions, model.deviceType, osType],
+		);
+		const versionSelectionOpts = useMemo(
+			() => (showAllVersions ? selectionOpts : preferredSelectionOpts),
+			[preferredSelectionOpts, selectionOpts, showAllVersions],
+		);
+		const showAllVersionsToggle = useMemo(
+			() => preferredSelectionOpts.length < selectionOpts.length,
+			[preferredSelectionOpts.length, selectionOpts.length],
+		);
+
+		const handleShowAllVersions = (e: React.ChangeEvent<HTMLInputElement>) => {
+			const isChecked = e.target.checked;
+			setShowAllVersions(isChecked);
+
+			if (isChecked || !version) {
+				return;
+			}
+
+			const selectedValueIsPreferred = preferredSelectionOpts.some(
+				(ver) => ver.value === version.value,
+			);
+			if (selectedValueIsPreferred) {
+				return;
+			}
+
+			const preferred =
+				preferredSelectionOpts.find((ver) => ver.isRecommended) ??
+				preferredSelectionOpts?.[0];
+			if (preferred) {
+				setVersion(preferred);
+			}
+		};
+
+		const handleSelectedDeviceTypeChange = useCallback(
+			(deviceType: DeviceType) => {
+				if (model.deviceType.slug === deviceType.slug) {
+					return;
+				}
+
+				const newDeviceType = compatibleDeviceTypes?.find(
+					(cdt) => cdt.slug === deviceType.slug,
+				);
+				if (!newDeviceType) {
+					return;
+				}
+				onSelectedDeviceTypeChange(newDeviceType);
+				setVersion(undefined);
+			},
+			[
+				compatibleDeviceTypes,
+				model.deviceType.slug,
+				onSelectedDeviceTypeChange,
+			],
+		);
+
+		useEffect(() => {
+			if (!version) {
+				const newVersion =
+					versionSelectionOpts.find((ver) => ver.isRecommended) ??
+					versionSelectionOpts[0];
+				onChange('version', newVersion?.value);
+				setVersion(newVersion);
+				return;
+			}
+
+			const versionWithVariant = version.hasPrebuiltVariants
+				? version.rawVersions[variant]
+				: version.rawVersion;
+			if (versionWithVariant) {
+				onChange('version', versionWithVariant);
+				onChange('developmentMode', variant === 'dev');
+			}
+
+			if (version.hasPrebuiltVariants && !version.rawVersions[variant]) {
+				setVariant(variant === 'dev' ? 'prod' : 'dev');
+			}
+		}, [version, variant, onChange, versionSelectionOpts]);
+
+		return (
+			<Grid
+				action={downloadUrl}
+				method="post"
+				component="form"
+				noValidate
+				autoComplete="off"
+				p={2}
+				ref={formElement}
+			>
+				<input type="hidden" name="_token" value={authToken} />
+				<input type="hidden" name="appId" value={applicationId} />
+				<input type="hidden" name="fileType" value=".zip" />
+				<Grid py={3} display="flex" flexWrap="wrap" gap={2}>
+					{compatibleDeviceTypes && compatibleDeviceTypes.length > 1 && (
+						<Grid display="flex" flexDirection="column" flex="1">
```

Comment:
Why are we using `Grid` here instead of `Box`? I don't see us using any of the alignment properties that Grid grants us

Also I see that this is Gridv1, not Gridv2 like as we agreed to stick to